### DISCUSSION TITLE
Docker Container & uhubctl

### DISCUSSION THREAD

**Discussion by LalunePierr0t** [mvp/uhubctl#487] [Q&A] [answered]
Hi 

Thanks for your work. 
For testing automation purpose, I plan to use uhuctl into a docker container to drive host usb hub .

Have you  ever done this ? 
Have you a guideline ?

Regards

**Comment by mvp** [comment, marked as answer]
I have not tried this. I doubt it will work. Usually, virtualization platforms only virtualize USB devices, but not hubs they are attached to. uhubctl needs access to USB hubs. The only way for this to work is if you can see hubs inside your guest VM.

**Comment by SafaZM** [comment]
Hi @LalunePierr0t 
using docker on linux, it is possible to use uhubctl in docker container. You have to provide the necessary volume (`-v /dev/bus/usb:/dev/bus/usb`) and providing privileged mode. 
My answer is late, but perhaps still interesting.
